stage1/init/init.go

```diff
@@ -110,8 +127,40 @@ func main() {
 	env = append(env, "LD_PRELOAD="+filepath.Join(path.Stage1RootfsPath(c.Root), "fakesdboot.so"))
 	env = append(env, "LD_LIBRARY_PATH="+filepath.Join(path.Stage1RootfsPath(c.Root), "usr/lib"))
 
-	if err := syscall.Exec(args[0], args, env); err != nil {
-		fmt.Fprintf(os.Stderr, "Failed to execute nspawn: %v\n", err)
-		os.Exit(5)
+	if privNet {
+		n, err := networking.Setup(c.Manifest.UUID)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to setup network: %v\n", err)
+			return 6
+		}
+		defer n.Teardown()
+
+		if err = n.EnterContNS(); err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to switch to container netns: %v\n", err)
+			return 6
+		}
+
+		cmd := exec.Command(args[0], args[1:]...)
+		cmd.Stdin = os.Stdin
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		cmd.Env = env
+		if err := cmd.Run(); err != nil {
```

Comment:
since the error handling is the same, set this err and check it outside of the branches?